### PR TITLE
carving: Adding distributed request_id to carver

### PR DESCRIPTION
**Pull request by poppyseedplehzr**
This adds the distributed request ID to the file carver so requests made by a distributed TLS endpoint can be connected on the endpoints.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 49f2ccd95e714e1c5be4ae2e69138e7e69f30c9b (Job results: [1486](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1486/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 49f2ccd95e714e1c5be4ae2e69138e7e69f30c9b (Job results: [4625](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4625/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 49f2ccd95e714e1c5be4ae2e69138e7e69f30c9b (Job results: [1511](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1511/)) failed one or more tests (**Windows**).

**Comment by poppyseedplehzr** [comment]
ok to test

File: include/osquery/database.h [hunk 1]
```diff
@@ -50,6 +50,9 @@ extern const std::string kQueries;
 /// The "domain" where event results are stored, queued for querytime retrieval.
 extern const std::string kEvents;
 
+/// The "domain" where the results of carve queries are stored.
+extern const std::string kCarves;
```
Comment (jedi22) [inline include/osquery/database.h]: @theopolis We good with adding new domains?

File: osquery/carver/carver.cpp [hunk 2]
```diff
@@ -108,7 +108,9 @@ void updateCarveValue(const std::string& guid,
   }
 }
 
-Carver::Carver(const std::set<std::string>& paths, const std::string& guid) {
+Carver::Carver(const std::set<std::string>& paths,
+               const std::string& guid,
+               const std::string& requestId) {
   status_ = Status(0, "Ok");
```
Comment (jedi22) [inline osquery/carver/carver.cpp]: Status(0);

File: osquery/tables/forensic/carves.cpp [hunk 3]
```diff
@@ -116,7 +117,8 @@ QueryData genCarves(QueryContext& context) {
       LOG(WARNING) << "Error inserting new carve entry into the database: "
                    << s.getMessage();
     } else {
-      Dispatcher::addService(std::make_shared<Carver>(paths, guid));
+      auto requestId = Distributed::getCurrentRequestId();
```
Comment (jedi22) [inline osquery/tables/forensic/carves.cpp]: Probably not an issue not but if we have multiple threads doing distributed queries, we could get two carves with the same requestID I think.

File: include/osquery/database.h (hunk 1 above)
Comment (poppyseedplehzr) [inline include/osquery/database.h]: I think the desire here was to keep carves separated out from traditional queries. I can move this back into `kQueries` if we'd like, but this was mostly just a residual issue from the first PR bringing the initial functionality. As it stands currently querying for all executed `carves` will give you nothing as the domain doesn't yet exist.

**Pull request merged**